### PR TITLE
Added "Use hooks to export state via environment variables" Section in setup.md

### PR DESCRIPTION
**Pull request by cptpiepmatz**
@amtoine meant that something like this could be a nice addition to the cookbook.

I personally use this as the example I gave in here.

### CODE REVIEW

File: cookbook/setup.md [hunk 1]
```diff
@@ -114,3 +114,43 @@ or
...
+
+```nu
+# set NU_OVERLAYS with overlay list, useful for starship prompt
+$env.config.hooks.pre_prompt = ($env.config.hooks.pre_prompt | append {||
+  let nu_overlays = (overlay list | range 1.. | str join ", ")
+  if ($nu_overlays | str length | into bool) {
+    $env.NU_OVERLAYS = $nu_overlays
+  } else {
+    $env.NU_OVERLAYS = null
+  }
```
Comment (amtoine) [inline cookbook/setup.md:144]: i might just propose the following change
```suggestion
  let overlays = overlay list | range 1..
  if not ($overlays | is-empty) {
    $env.NU_OVERLAYS = $overlays | str join ", "
  } else {
    $env.NU_OVERLAYS = null
  }
```
i find it a bit clearer without the `str length | into bool` but i'll let you decide :yum: 
it's really not a strong request at all :wink:

Comment (cptpiepmatz) [reply cookbook/setup.md:144]: It is way more clear what the intention is, so this is nice 👍🏻

Comment (amtoine) [reply cookbook/setup.md]: thanks for the quick response :blush:

**Review by amtoine — APPROVED** [review]
thanks for this @cptpiepmatz :pray: 

let's run the CI for good measure and then land this PR :muscle:

**Comment by amtoine** [comment]
it's all :green_circle:, let's gooooo :partying_face:

**Pull request merged**